lib/saluki-config/src/refresher.rs

```diff
@@ -0,0 +1,206 @@
+use std::sync::Arc;
+
+use arc_swap::ArcSwap;
+use reqwest::ClientBuilder;
+use saluki_error::GenericError;
+use serde::{de::DeserializeOwned, Deserialize};
+use serde_json::Value;
+use tokio::time::{sleep, Duration};
+use tracing::{debug, error};
+
+use crate::{ConfigurationError, GenericConfiguration};
+
+const DEFAULT_AGENT_IPC_HOST: &str = "localhost";
+const DEFAULT_AUTH_TOKEN_FILE_PATH: &str = "/etc/datadog-agent/auth_token";
+const DEFAULT_REFRESH_INTERVAL_SECONDS: u64 = 15;
+
+/// Configuration for setting up `RefreshableConfiguration`.
+#[derive(Default, Deserialize)]
+pub struct RefresherConfiguration {
+    /// The location of the auth token used by the Datadog Agent.
+    ///
+    /// Defaults to `/etc/datadog-agent/auth_token`.
+    #[serde(default = "default_auth_token_file_path")]
+    auth_token_file_path: String,
+
+    /// The amount of time betweeen each request in seconds.
+    ///
+    /// Defaults to 15 seconds.
+    #[serde(
+        rename = "agent_config_refresh_internal_seconds",
+        default = "default_refresh_interval_seconds"
+    )]
+    refresh_interval_seconds: u64,
+
+    /// The IPC host used by the Datadog Agent.
+    ///
+    /// Defaults to `localhost`.
+    #[serde(default = "default_agent_ipc_host")]
+    agent_ipc_host: String,
+
+    /// The IPC port used by the Datadog Agent.
+    ///
+    /// Defaults to `0`.
+    #[serde(default = "default_agent_ipc_port")]
+    agent_ipc_port: u64,
+}
+
+fn default_auth_token_file_path() -> String {
+    DEFAULT_AUTH_TOKEN_FILE_PATH.to_owned()
+}
+
+fn default_refresh_interval_seconds() -> u64 {
+    DEFAULT_REFRESH_INTERVAL_SECONDS
+}
+
+fn default_agent_ipc_host() -> String {
+    DEFAULT_AGENT_IPC_HOST.to_owned()
+}
+
+fn default_agent_ipc_port() -> u64 {
+    0
```

Comment:
Zero still doesn't make sense as a value for something we're connecting _to_, so we shouldn't specify a default port at all. This would let us bubble up an error when calling `RefresherConfiguration::from_configuration` if `agent_ipc_port` isn't set.

Probably have the code building the refreshable config in `bin/agent-data-plane/src/main.rs` catch the error and report an INFO-level log that dynamic configuration refreshing will be unavailable due to not being configured, etc etc.